web_poet/page_inputs.py

```diff
@@ -53,18 +74,20 @@ def from_name_value_pairs(cls, arg: List[Dict]):
         """
         return cls([(pair["name"], pair["value"]) for pair in arg])
 
+    def declared_encoding(self) -> Optional[str]:
+        """ Return encoding detected from the Content-Type header, or None
+        if encoding is not found """
+        content_type = self.get('Content-Type', '')
+        return http_content_type_encoding(content_type)
+
 
-@attr.define
-class ResponseData:
+@attrs.define(auto_attribs=False)
+class HttpResponse:
```

Comment:
Renaming this from `ResponseData` into `HttpResponse` would warrant the `status` and `headers` to be **non-optional** for them to semantically represent a proper HTTP Response.

Reply:
Yeah, that's a good call out. I left them optional, and default to None mostly because of an example you have in the docs (loading html files from disk). Status and headers are not needed for extraction in many cases.

I was also thinking about making them required, but providing defaults: 200 status code, empty headers. This has a large advantage: user code doesn't need to handle "response.headers is None" case, response.headers.get("content-type") can safely be called. At the same time, code to create responses remain simple. A downside is that you can't know if 200 is a real status code, or a status code is not available; the same for headers.

Reply:
What do you think about option 4?

```py
status: Optional[int] = attrs.field(default=None)
headers: HttpResponseHeaders = attrs.field(factory=HttpResponseHeaders)
```

I.e. allow status to be None, but not headers - provide "fake" empty headers by default?

Reply:
Good idea. This is slightly better than option 1 as a header instance is already available. This prevents `response.headers.get("Content-Type")` from unintentionally breaking compared to being `None`.

Reply:
Implemented it.